### PR TITLE
meson: fix install path of example .network files

### LINKED ISSUE #29925: Example `.network` files are installed to an incorrect directory
I just randomly came across this while debugging something else - it looks like the couple of example `.network` files gets installed into `/usr/lib/systemd/network/network/` when `-Ddefault-network=true` is used:

```
# DESTDIR=$PWD/root ninja -C build install
# ls -lR root/usr/lib/systemd/network/ 
root/usr/lib/systemd/network/:
total 36
-rw-r--r--. 1 root root  819 Nov  8 04:29 80-6rd-tunnel.network
-rw-r--r--. 1 root root  937 Nov  8 04:29 80-container-host0.network
-rw-r--r--. 1 root root  942 Nov  8 04:29 80-container-vb.network
-rw-r--r--. 1 root root 1039 Nov  8 04:29 80-container-ve.network
-rw-r--r--. 1 root root 1025 Nov  8 04:29 80-container-vz.network
-rw-r--r--. 1 root root  986 Nov  8 04:29 80-vm-vt.network
-rw-r--r--. 1 root root  730 Nov  8 04:29 80-wifi-adhoc.network
-rw-r--r--. 1 root root  769 Nov  8 04:29 99-default.link
drwxr-xr-x  2 root root 4096 Nov  8 04:59 network

root/usr/lib/systemd/network/network:
total 16
-rw-r--r--. 1 root root 719 Nov  8 04:29 80-auto-link-local.network
-rw-r--r--. 1 root root 664 Nov  8 04:29 80-wifi-ap.network
-rw-r--r--. 1 root root 595 Nov  8 04:29 80-wifi-station.network
-rw-r--r--. 1 root root 619 Nov  8 04:29 89-ethernet.network
```

Compare it with:
```
# DESTDIR=$PWD/root-false ninja -C build-false install
# ls -lR root-false/usr/lib/systemd/network/ 
root-false/usr/lib/systemd/network/:
total 48
-rw-r--r--. 1 root root  819 Nov  8 04:29 80-6rd-tunnel.network
-rw-r--r--. 1 root root  719 Nov  8 04:29 80-auto-link-local.network.example
-rw-r--r--. 1 root root  937 Nov  8 04:29 80-container-host0.network
-rw-r--r--. 1 root root  942 Nov  8 04:29 80-container-vb.network
-rw-r--r--. 1 root root 1039 Nov  8 04:29 80-container-ve.network
-rw-r--r--. 1 root root 1025 Nov  8 04:29 80-container-vz.network
-rw-r--r--. 1 root root  986 Nov  8 04:29 80-vm-vt.network
-rw-r--r--. 1 root root  730 Nov  8 04:29 80-wifi-adhoc.network
-rw-r--r--. 1 root root  664 Nov  8 04:29 80-wifi-ap.network.example
-rw-r--r--. 1 root root  595 Nov  8 04:29 80-wifi-station.network.example
-rw-r--r--. 1 root root  619 Nov  8 04:29 89-ethernet.network.example
-rw-r--r--. 1 root root  769 Nov  8 04:29 99-default.link
```

Looks like the `foreach` or `replace_suffix` messes things up:
https://github.com/systemd/systemd/blob/e3c58427323bbecfcef6c7f01b8f43b1a30eb6e7/network/meson.build#L22-L26

/cc @yuwata

### PR DESCRIPTION
**Pull request by yuwata**
It seems that when 'rename' field is set, the path (instead of the filename) is appended to the 'install_dir'.

Follow-up for 9b7a624267fddc5c20bd15480e7a393d7a3b270e.

Fixes #29925.

split-out of #29928, to make this merged earlier.

### CODE REVIEW

**Review by mrc0mmand — APPROVED** [review]

**Comment by bluca** [comment]
Sorry but no - the fact that the path was broken is the only thing that saved me from shipping these by default, which is not right, so the two things need to be changed together

**Comment by poettering** [comment]
I am sorry, what?  you think it's better to install this in the wrong path?

**Comment by poettering** [comment]
Maybe cool down for a minute, man. You are not thinking straight.

**Comment by yuwata** [comment]
@bluca The purpose of this PR is for splitting out pure fix part. I hope no one disagrees this change. Let's continue discussion about the default value in another PR.

**Comment by bluca** [comment]
> @bluca The purpose of this PR is for splitting out pure fix part. I hope no one disagrees this change. Let's continue discussion about the default value in another PR.

But I do disagree with that, because it means they do get silently enabled by default, which as already mentioned, is not good.

**Comment by yuwata** [comment]
> > @bluca The purpose of this PR is for splitting out pure fix part. I hope no one disagrees this change. Let's continue discussion about the default value in another PR.
> 
> But I do disagree with that, because it means they do get silently enabled by default, which as already mentioned, is not good.

This does not intend to close nor replace the other PR... I just want to merge fix for the trivial bug.

**Comment by poettering** [comment]
> But I do disagree with that, because it means they do get silently enabled by default, which as already mentioned, is not good.

It's not very nice to block this technical fix for something that is already in place due to a philosophical disagreement on something not immediately related to the technical fix.

I understand you have some strong opinions, and don#t want to read NEWS, but I am not convinced this should block fixing obvious bugs.

**Comment by mrc0mmand** [comment]
I don't really care how this is going to be resolved, but in its current form this breaks the CI machines, since the defaults are picked up by networkd and it fucks up networking.

**Comment by bluca** [comment]
> > But I do disagree with that, because it means they do get silently enabled by default, which as already mentioned, is not good.
> 
> It's not very nice to block this technical fix for something that is already in place due to a philosophical disagreement on something not immediately related to the technical fix.
> 
> I understand you have some strong opinions, and don#t want to read NEWS, but I am not convinced this should block fixing obvious bugs.

The CI is red

**Comment by yuwata** [comment]
Now all green.
Yeah, enabling to install the default .network files actually causes several issues on our CIs. But the point is out of the scope of this PR, and should be discussed in the other PR. And the tests should work out of the box anyway even if the feature is enabled.
Please take a look.

**Comment by bluca** [comment]
ubuntu CI is still down, should be back up soon

**Comment by yuwata** [comment]
Oh, indeed. I did not notice that. That's why most PRs are all green. Haha...
@bluca Will you (re-)start Ubuntu CIs on this PR? Or should I force-push later again?

**Comment by keszybz** [comment]
Yeah, this is very problematic, since fixing the path would break systems. Let's finish the discussion in https://github.com/systemd/systemd/pull/29928 first.

**Comment by yuwata** [comment]
> Yeah, this is very problematic, since fixing the path would break systems. Let's finish the discussion in #29928 first.

Again, this is not a replacement of #29928. But, one technical fix and test update to make them happy even -Ddefault-network is enabled.
Please merge this before #29928. I will rebase #29928 after this is merged.

Now, all tests are green (though, ubuntu CIs are not available, but networkd-tests.py is tested also on semaphoreci). So, we can merge this now. PTAL!!

**Comment by yuwata** [comment]
In other words, this is completely independent of the direction of the discussion in #29928.

**Comment by keszybz** [comment]
> In other words, this is completely independent of the direction of the discussion in #29928.

It's not. Both the Fedora package and the Debian package maintainers missed the implications of this change and the users of those distributions were not unexpectedly affected only because of the mistake in file paths. The same also applies to other users.

**Comment by yuwata** [comment]
> > In other words, this is completely independent of the direction of the discussion in #29928.
> 
> It's not. Both the Fedora package and the Debian package maintainers missed the implications of this change and the users of those distributions were not unexpectedly affected only because of the mistake in file paths. The same also applies to other users.

I cannot follow. This is independent of packaging...

**Comment by mrc0mmand** [comment]
> > > In other words, this is completely independent of the direction of the discussion in #29928.
> > 
> > 
> > It's not. Both the Fedora package and the Debian package maintainers missed the implications of this change and the users of those distributions were not unexpectedly affected only because of the mistake in file paths. The same also applies to other users.
> 
> I cannot follow. This is independent of packaging...

I think what @keszybz is saying is that the example .network files were picked up by the respective distro packages (probably, because they package everything systemd installs under /usr/lib/systemd/network/), but "thanks" to the bug in the installation path i.e. /usr/lib/systemd/network/network/) it didn't screw up users' network setups (like it did in CI when the installation path got fixed).

**Comment by yuwata** [comment]
> > > > In other words, this is completely independent of the direction of the discussion in #29928.
> > > 
> > > 
> > > It's not. Both the Fedora package and the Debian package maintainers missed the implications of this change and the users of those distributions were not unexpectedly affected only because of the mistake in file paths. The same also applies to other users.
> > 
> > 
> > I cannot follow. This is independent of packaging...
> 
> I think what @keszybz is saying is that the example .network files were picked up by the respective distro packages (probably, because they package everything systemd installs under /usr/lib/systemd/network/), but "thanks" to the bug in the installation path i.e. /usr/lib/systemd/network/network/) it didn't screw up users' network setups (like it did in CI when the installation path got fixed). But if we merged this PR as is, i.e. with shipping the .network files by default, this would then might cause issues in the wild.

Still cannot follow. The packaging thing is already tracked and discussed in #29928. We can merge #29928 after this PR but before -RC2 if necessary. Why this PR needs to take care of packaging?

**Comment by YHNdnzj** [comment]
Yeah, let's not mix two questions here. This is a bug, whether we disable the option by default or not. The other PR should be a RC blocker, but this PR should really not get blocked.

**Pull request merged**
